### PR TITLE
#156615498 restrict room addition

### PR DESCRIPTION
**Pull request by proxiex**
#### What does this PR do?
- restricts room addition to admin only
#### Description of Task to be completed?
- restricts room addition to admin only
-- decode token
-- check if user is listed as admin
-- show error message if necessary

#### How should this be manually tested?
- add @user_role('role_here') decoretor the the fuction needed to be protected
- run app and visit **http://127.0.0.1:5000/mrm/
- pass token to the header
- make request

#### Any background context you want to provide?
n/n
#### What are the relevant pivotal tracker stories?
[156615498](https://www.pivotaltracker.com/n/projects/2154921)
#### Screenshots (if appropriate)
screen shot 2018-05-14 at 10 17 20 am

#### Questions:
n/a

### CODE REVIEW

File: api/role/schema.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import graphene
+from graphene_sqlalchemy import ( 
+    SQLAlchemyObjectType, 
+    SQLAlchemyConnectionField
```
Comment (PatrickCmd) [inline api/role/schema.py]: I think *SQLAlchemyConnectionField* since we are no longer using relay interfaces

File: api/user/schema.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
+import graphene
+
+from graphene import relay, Schema
+from graphene_sqlalchemy import (SQLAlchemyObjectType, 
+                                 SQLAlchemyConnectionField)
```
Comment (PatrickCmd) [inline api/user/schema.py]: Same as above.

File: helpers/decorators/auth.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        try:
+            payload = jwt.decode(auth_token, SECRET_KEY)
+            return payload
+        except jwt.ExpiredSignatureError:
+            return jsonify({
+                'message': 'Signature expired. Please log in again.'}), 401
+        except jwt.InvalidTokenError:
+            return jsonify({
+                'message': 'Invalid token. Please Provied a valid token!'
```
Comment (PatrickCmd) [inline helpers/decorators/auth.py]: You have typo **Provied**, I think you meant **Provide**

File: helpers/decorators/auth.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        return wrapper
+
+    def save_user(self, payload):
+        """
+        Save user to database.
+        
+        params: 
+            payload: dict
+        returns: 
+            bloolean
```
Comment (PatrickCmd) [inline helpers/decorators/auth.py]: Typo **bloolean**, I think you meant **boolean**

File: api/role/schema.py (hunk 1 above)
Comment (proxiex) [reply api/role/schema.py]: 🤭Oops, good catch! I'll fix it.

File: helpers/decorators/auth.py (hunk 3 above)
Comment (proxiex) [reply helpers/decorators/auth.py]: Thank you.

File: helpers/decorators/auth.py (hunk 4 above)
Comment (proxiex) [reply helpers/decorators/auth.py]: Will correct it

File: alembic/env.py [hunk 5]
```diff
@@ -19,24 +19,20 @@
...
-import os
-import sys
-sys.path.append(os.getcwd())
-=======
 # target_metadata = None
 
 sys.path.append(os.getcwd())
 config.set_main_option('sqlalchemy.url', os.getenv('DEV_DATABASE_URL'))
 
->>>>>>> 886bd3f4174947f2fb32aba10b8f635d69cbd92a
```
Comment (shemogumbe) [inline alembic/env.py]: This seems to be inappropriate code, please this was fixed, have you rebased this branch with latest develop?

File: app.py [hunk 6]
```diff
@@ -13,13 +12,18 @@ def create_app(config_name):
     app.config.from_object(config[config_name])
     config[config_name].init_app(app)
     
-    app.add_url_rule(
-        '/mrm',
-        view_func=GraphQLView.as_view(
+    def graphql_view():
```
Comment (shemogumbe) [inline app.py]: I am against authenticating routes in graphQL, remember we only have one route, and there are things we would need no authentication for, why not just go the route of authorization

**Comment by JoyLubega** [comment]
Proxie, I completed the authentication part. You don't need to redo it..

**Pull request merged**